src/workloads/sharding/WouldChangeOwningShardBatchWrite.yml

```diff
@@ -0,0 +1,91 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/sharding"
+Description: |
+  Creates a sharded collection with 2 chunks on 2 different shards using ranged sharding and updates
+  the shard key to trigger WouldChangeOwningShard errors.
+
+  The workload consists of 3 phases:
+    1. Shard an empty collection (using ranged sharding) into two chunks and wait for the balancer 
+    to distribute each chunk to its own shard.
+    2. Populate the sharded collection with data.
+    3. Update the shard key value to trigger WouldChangeOwningShard errors.
+
+GlobalDefaults:
+  Nop: &Nop {Nop: true}
+
+  Database: &Database test
+  # Collection0 is the default collection populated by the MonotonicSingleLoader.
+  Collection: &Collection Collection0
+  Namespace: &Namespace test.Collection0
+
+  DocumentCount: &DocumentCount 1000  # Number of documents to insert and modify.
+  BalancerWait: &BalancerWait "15 seconds" # Wait out the 10 second delay between balancing rounds
+  # with a little extra leeway for balancing to finish.
+
+Actors:
+- Name: CreateShardedCollection
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: admin
+    Operations:
+    - OperationMetricsName: EnableSharding
+      OperationName: AdminCommand
+      OperationCommand:
+        enableSharding: *Database
+    - OperationMetricsName: ShardCollection
+      OperationName: AdminCommand
+      OperationCommand:
+        shardCollection: *Namespace
+        key: {x: 1}
+    - OperationMetricsName: SplitChunks
+      OperationName: AdminCommand
+      OperationCommand:
+        split: *Namespace
+        middle: {x: 5}
+  - *Nop
+  - *Nop
+
+- Name: LoadInitialData
+  Type: MonotonicSingleLoader
+  Threads: 1
+  Phases:
+  - *Nop
+  - Repeat: 1
+    SleepBefore: *BalancerWait
+    BatchSize: 1000
+    DocumentCount: *DocumentCount
+    Database: *Database
+    Document: {x: 0}
+  - *Nop
+
+- Name: UpdateShardKey
```

Comment:
From https://github.com/mongodb/mongo/blob/master/src/mongo/s/commands/cluster_write_cmd.cpp#L152-L159 If we send a retryable write command and it fails with WouldChangeOwningShard, we implicitly retry in a transaction.

Reply:
Does genny use sessions by default? According to this, it looks like you need to specify `OnSession: true`. `retryableWrite` defaults to true so I think the operation will run as a retryable write by default after you specify `OnSession: true`.

https://github.com/mongodb/genny/blob/446f8ebdc10fb1d8d7d0a940887ef1b7a4203f71/src/workloads/docs/CrudActorTransaction.yml#L38-L40 

Also, could you add a phrase that runs the update inside a transaction? I think you can use `withTransaction` to do that. Here is an example: https://github.com/mongodb/genny/blob/446f8ebdc10fb1d8d7d0a940887ef1b7a4203f71/src/workloads/docs/CrudActorTransaction.yml#L12-L40

According to the source code for `WithTransactionOperation`, it corresponds to `with_transaction` in the driver API http://mongocxx.org/api/current/classmongocxx_1_1client__session.html#a2ca899e8c0018af269f2542b3538587f.

Reply:
As discussed offline, upon getting WouldChangeOwningShard for a retryable write, we automatically retry it in a transaction. So adding a new phase for running it in a transaction is a little redundant. I think retryableWrites are enabled by default in drivers: https://docs.mongodb.com/v4.0/core/retryable-writes/ .